RELEASE.md

```diff
@@ -34,4 +36,39 @@ resolvers ++= Seq(
   )
 ```
 
-## TODO(tomerk): Add details about publishing a full release
+## Full Release
+
+1. Update the build.sbt and the Makefile to the new version
+
+2. Cut a release branch for major version releases
+
+3. Tag the specific commit of the release branch to be released with its version
+
+4. We publish KeystoneML releases and snapshots to sonatype repository https://oss.sonatype.org/. To
+create a release you will first need to create a SonaType account.
+
+5. Create and distribute a gpg key if you don't have one. The steps to do that are:
+```
+  gpg --gen-key # Follow the prompts to create a key
+  # Publish the key to a keyserver. If this doesn't work go to https://sks-keyservers.net/i/
+  # and paste the output of gpg --armor --export <email-id-used-in-previous-step>
+  gpg --send-keys
+```
+
+6. Add `sbt-pgp` and `sbt-sonatype` to your plugins at `~/.sbt/0.13/plugins/gpg.sbt`
+```
+addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.0")
+
+addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "1.1")
+```
+
+7. Add your Sonatype credentials to `~/.sbt/0.13/sonatype.sbt`
+credentials += Credentials("Sonatype Nexus Repository Manager",
+                           "oss.sonatype.org",
+                           "<userame>",
+                           "<password>")
+
+8. Run `sbt/sbt publishSigned` to upload to sonatype a staging repository.
+
+9. Follow [these instructions](http://central.sonatype.org/pages/releasing-the-deployment.html) to
```

Comment:
Can we make a note to try this for the 0.3 release and see if it works ? It'll be cool to not have to mess with the web UI for just the promoting bits.